src/bin/pg_autoctl/keeper_config.h

```diff
@@ -48,6 +48,9 @@ typedef struct KeeperConfig
 	int prepare_promotion_walreceiver;
 	int postgresql_restart_failure_timeout;
 	int postgresql_restart_failure_max_retries;
+
+	/* pg_autoctl show uri */
+	bool monitorUriOnly;
```

Comment:
We don't add command line option switches to our internal state data structures. That's why we have global variables for some switches (like outputJSON for instance, and keeperOptions and monitorOptions. It would be great to have a different design with a command line specific structure (that could still have a PostgresSetup field), but that's not what we have now and it might be a little too much to ask for this PR to refactor the whole CLI option business...

Still, I think it's important that we refrain from adding command-line processing features to the keeper state data structures.

Reply:
Created a `ShowUriOptions` struct to store the options now.